### PR TITLE
Merge 2.12.x to 2.13.x [ci: last-only]

### PR DESCRIPTION
**Pull request by retronym**
```
% git co origin/2.13.x; export MB=$(git merge-base head origin/2.12.x); git log --graph --oneline $MB..origin/2.12.x | tail -r
HEAD is now at 4bfe574938 Merge pull request #7128 from retronym/perf/find-dir-entry
* ab99db089b Rewrite List(a, b) to `new ::(a, new ::(b, Nil)`
* 296e3167a8 add a simple test for optimised List.apply rewrite in cleanup
* 9a2219405b Merge pull request #6994 from rorygraves/mike/2.12.x_ListApply
| * 0763168ca6 [nomerge] Emit bridge method forwarders with BRIDGE flag
|\
*   4c9780fb7a Merge pull request #7035 from eed3si9n/bport/b10812
|/
| * 49a5ebde96 Uncurry of Java varargs respects Java's universal trait nescience
|\
*   15f75ad98d Merge pull request #7141 from hrhino/t11109
| * 42dac30eb5 [nomerge] fix performance regression in mutable.HashMap#getOrElseUpdate
|\
*   2b2356e70f Merge pull request #7091 from SethTisue/mutable-hashmap-getorelseupdate
| * aa7578210b Don't re-apply type maps to info of unchanged symbols
| * 228d21780d Make internal methods in TypeMap more private/final/inlinable
|\
*   38282c7af1 Merge pull request #7137 from hrhino/faster/type-map

% git merge 9a2219405b
Auto-merging src/reflect/scala/reflect/internal/Definitions.scala
CONFLICT (content): Merge conflict in src/reflect/scala/reflect/internal/Definitions.scala
Auto-merging src/compiler/scala/tools/nsc/transform/CleanUp.scala
CONFLICT (content): Merge conflict in src/compiler/scala/tools/nsc/transform/CleanUp.scala
Automatic merge failed; fix conflicts and then commit the result.

% git merge -sours 4c9780fb7a
Merge made by the 'ours' strategy.

% git merge -sours 2b2356e70f
Merge made by the 'ours' strategy.

% git merge 38282c7af1
Auto-merging src/reflect/scala/reflect/internal/tpe/TypeMaps.scala
CONFLICT (content): Merge conflict in src/reflect/scala/reflect/internal/tpe/TypeMaps.scala
Automatic merge failed; fix conflicts and then commit the result.
```

### CODE REVIEW

**Comment by lrytz** [comment]
```
sbt.ForkMain$ForkError: java.lang.ClassCastException: scala.collection.immutable.$colon$colon cannot be cast to scala.collection.immutable.Set
	at scala.reflect.internal.StdNames$TermNames.<init>(StdNames.scala:952)
```

**Comment by retronym** [comment]
Pushed a commit to fix the semantic merge error.

**Comment by SethTisue** [comment]
```
[test] !!  235 - run/list-apply-eval.scala                 [output differs]
[test] % diff /home/jenkins/workspace/scala-2.13.x-validate-main/test/files/run/list-apply-eval.check /home/jenkins/workspace/scala-2.13.x-validate-main/test/files/run/list-apply-eval-run.log--- empty
[test] +++ list-apply-eval-run.log
[test] @@ -1,0 +1,1 @@
[test] +warning: there was one deprecation warning (since 2.13.0); re-run with -deprecation for details
[test] ##### Log file '/home/jenkins/workspace/scala-2.13.x-validate-main/test/files/run/list-apply-eval-run.log' from failed test #####
[test] 
[test] warning: there was one deprecation warning (since 2.13.0); re-run with -deprecation for details
[test] 
```

```
[test] !! 26 - run/t191.scala                            [output differs][duration 5.56s]
[test] % diff /home/jenkins/workspace/scala-2.13.x-validate-main/test/scaladoc/run/t191.check /home/jenkins/workspace/scala-2.13.x-validate-main/test/scaladoc/run/t191-run.log
[test] @@ -1 +1,48 @@
[test] -Done.
[test] +java.lang.AssertionError: assertion failed
```

File: src/reflect/scala/reflect/internal/Definitions.scala [hunk 1]
```diff
@@ -1587,7 +1587,21 @@ trait Definitions extends api.StandardDefinitions {
       lazy val Boolean_not = definitions.Boolean_not
 
       lazy val Option_apply = getMemberMethod(OptionModule, nme.apply)
-      lazy val List_apply = DefinitionsClass.this.List_apply
+      private lazy val List_apply = DefinitionsClass.this.List_apply
```
Comment (lrytz) [inline src/reflect/scala/reflect/internal/Definitions.scala:1590]: The field name `List_apply` seems quite misleading, should we rename it to `IterableFactory` (and also resolve as such)?

**Review by lrytz — APPROVED** [review]
LGTM, we can fix the field name post factum

**Comment by lrytz** [comment]
Argh, it seems https://github.com/scala/scala/pull/7141 got merged with `-s ours` by mistake - or did I miss something?

**Pull request merged**